cypress/integration/chi-js/modal.spec.js

```diff
@@ -0,0 +1,182 @@
+describe('chi-modal', function() {
+  before(() => {
+    cy.visit('http://localhost:8000/tests/js/modal.html');
+  });
+
+  describe('check base modal', () => {
+    it(`modal should be opened when we click launch demo modal button`, () => {
+      cy.get('button#modal-trigger-1').click();
+      cy.get('#modal-1')
+        .should('have.class', '-active')
+        .find('.-close')
+        .click()
+        .wait(550);
+    });
+
+    it(`modal should be closed when we click close button`, () => {
+      cy.get('button#modal-trigger-1').click();
+      cy.get('#modal-1')
+        .find('.-close')
+        .click()
+        .wait(550)
+        .should('be.not.visible');
+    });
+
+    it(`modal should be closed when we click cancel button`, () => {
+      cy.get('button#modal-trigger-1').click();
+      cy.get('#modal-1')
+        .contains('Cancel')
+        .click()
+        .wait(550)
+        .should('be.not.visible');
+    });
+  });
+
+  describe('check Center modal', () => {
+    it(`modal should be open in the center of the page when we click launch demo modal button`, () => {
+      cy.get('#modal-trigger-2').click();
+      cy.get('#modal-2')
+        .parent()
+        .parent()
+        .should('have.class', '-center')
+        .find('.-close')
+        .wait(550)
+        .click();
+    });
+  });
+
+  describe('check scrollable modal', () => {
+    it(`modal should be scrollable`, () => {
+      cy.get('button#modal-trigger-3')
+        .click()
+        .wait(550);
+      cy.get('#modal-3 .chi-modal__content')
+        .should('have.class', '-mh--400')
+        .scrollTo('bottom');
+      cy.get('#modal-3')
+        .find('.-close')
+        .click()
+        .wait(550)
+        .should('be.not.visible');
+    });
+  });
+
+  describe('check scrollable modal for long content', () => {
+    it(`modal should be scrollable when long content is present`, () => {
+      cy.get('button#modal-trigger-4')
+        .click()
+        .wait(550);
+      cy.get('#modal-trigger-4 + .chi-backdrop')
+        .should('have.class', '-mh--480')
+        .scrollTo('bottom');
+      cy.get('#modal-4')
+        .find('.-close')
+        .click()
+        .wait(550)
+        .should('be.not.visible');
+    });
+  });
+
+  describe('check modal with subtitle', () => {
+    it(`modal should contain subtitle`, () => {
+      cy.get('button#modal-trigger-5').click();
+      cy.get('#modal-5')
+        .find('.chi-modal__header')
+        .children()
+        .should('have.class', 'chi-modal__subtitle');
+      cy.get('#modal-5')
+        .find('.-close')
+        .click()
+        .wait(550);
+    });
+  });
+
+  describe('check modal with inverse', () => {
```

Comment:
Opening and closing will work the same as in the case of Base Modal.
The difference is the backdrop color which is checked by visual regression backstop tests. Let's remove the test case.